go/vt/worker/diff_utils_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package worker
+
+import (
+	"reflect"
+	"testing"
+
+	mproto "github.com/youtube/vitess/go/mysql/proto"
+	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/vt/key"
+	myproto "github.com/youtube/vitess/go/vt/mysqlctl/proto"
+)
+
+func TestOrderedColumns(t *testing.T) {
+	input := &myproto.TableDefinition{
+		PrimaryKeyColumns: []string{"pk1", "pk2"},
+		Columns:           []string{"pk1", "col1", "pk2", "col2"},
+	}
+	want := []string{"pk1", "pk2", "col1", "col2"}
+	if got := orderedColumns(input); !reflect.DeepEqual(got, want) {
+		t.Errorf("got %v, want %v", got, want)
+	}
+}
+
+func TestUint64FromKeyspaceId(t *testing.T) {
+	table := map[string]string{
```

Comment:
Generally if I see a table-based test that defines a map[string]string, I would assume the map key is the input and the value is the desired result. The "for input, want := range table" loop just below the table also makes this clear. In cases that aren't as simple as map[string]string being input->want, I would use a struct, like for TestCompareRows below, in which the field names describe the meaning.